### ISSUE TITLE
Oil spills generate during decoration instead of before

### ISSUE THREAD

**Issue by QuantumStatement** [BuildCraft/BuildCraft#4002]
BuildCraft version: 7.99.14
Forge version: 14.23.1.2604
Link to crash report or log: 
Singleplayer or multiplayer: Singleplayer
Additional information:
![2018-01-25_19 05 23](https://user-images.githubusercontent.com/35406698/35442458-60b1bc6e-02b8-11e8-966a-4930defa4f13.png)

**Comment by AlexIIL** [comment]
Yeah, oil generation really needs to happen *before* tree generation, or at last ignore leaves and logs when generating. Perhaps also removing the logs and leaves would be a good idea, if we cannot generate oil beforehand,

**Comment by QuantumStatement** [comment]
At least oil generation should ignore too large height difference, otherwise it will look like this
![2018-01-26_17 11 17](https://user-images.githubusercontent.com/35406698/35443649-e4b8b7de-02bc-11e8-80a4-f9790d3e4938.png)

**Comment by jburkej1** [comment]
I came across a village with an oil geyser in the middle, nearly every building was covered and oil was coming out of the church steeple. I regret not getting a screen shot. I cheated and used creative mode to setup pumps and tanks to siphon it off.

**Comment by LordGolias** [comment]
I will try to address this.

@AlexIIL, is there any simple way of easily testing oil generation? E.g. easily start a map with an oil on where the player starts?

**Comment by AlexIIL** [comment]
@LordGolias normally I generate random worlds (and explore for about a minute at max render distance) until I can see an oil spot that needs to be fixed. Then, note down the coordinates and turn them into a teleport command (like "/tp AlexIIL 345 98 201") and just paste that command in every time I regenerate the world (usually by doing "re-create" rather than "new world" as that keeps the seed the same).

Alternatively you could add a system.out.println statement to the spout generation code to tell you the coordinates every time a well generates.

I don't know of a good way of forcing the player to generate near an oil well, other than increasing the gen rate by a huge amount.
